### PR TITLE
Fix wso2/product-ei#2939

### PR DESCRIPTION
**Pull request by pramodya1994**
## Purpose

> Created dataservices.capp.deployer package to handle deploying data services using capps.
org.wso2.carbon.data:org.wso2.carbon.dataservices.capp.deployer bundle added to org.wso2.carbon.dataservices.server.feature.

## Related PRs

- https://github.com/wso2/carbon-kernel/pull/1900
- https://github.com/wso2/carbon-commons/pull/333

## Test environment

- JDK version: 1.8.0_181
- Operating system: Mac OS High Sierra

### CODE REVIEW

File: components/data-services/org.wso2.carbon.dataservices.capp.deployer/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                            !org.wso2.carbon.dataservices.capp.deployer.internal,
+                            org.wso2.carbon.dataservices.capp.deployer.*;version="${project.version}"
+                        </Export-Package>
+                    </instructions>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (indikasampath2000) [inline components/data-services/org.wso2.carbon.dataservices.capp.deployer/pom.xml]: should add a new line at the end of the file.

Comment (pramodya1994) [reply components/data-services/org.wso2.carbon.dataservices.capp.deployer/pom.xml]: Added

File: components/data-services/org.wso2.carbon.dataservices.capp.deployer/src/main/java/org/wso2/carbon/dataservices/capp/deployer/DataServiceCappDeployer.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    /**
+     * Deploy or un-deploy data services.
+     * if deploying, adding the data service to the data services.
+     * if un-deploying, removing the data service from data services.
+     * there can be multiple data sources as separate xml files.
+     *
+     * @param deploy    identify whether deployment process or un-deployment process.
+     * @param artifacts list of artifacts to be deployed.
+     */
+    private void deployUnDeployDataSources(boolean deploy, List<Artifact> artifacts, AxisConfiguration axisConfig)
```
Comment (sdkottegoda) [inline components/data-services/org.wso2.carbon.dataservices.capp.deployer/src/main/java/org/wso2/carbon/dataservices/capp/deployer/DataServiceCappDeployer.java]: I suppose we should not have a common method for both deploying and undeploying datasources. We know which method is being called at the compilation time itself. Hence shall we extract the common code to aseparate method and break the 2 logics into 2 methods?

Comment (pramodya1994) [reply components/data-services/org.wso2.carbon.dataservices.capp.deployer/src/main/java/org/wso2/carbon/dataservices/capp/deployer/DataServiceCappDeployer.java]: Before creating this class discussed this and decided to create this class using the same structure of DataSourceCappDeployer class.

Comment (pramodya1994) [reply components/data-services/org.wso2.carbon.dataservices.capp.deployer/src/main/java/org/wso2/carbon/dataservices/capp/deployer/DataServiceCappDeployer.java]: Fixed in https://github.com/wso2/carbon-data/pull/254/files

**Pull request merged**